### DISCUSSION TITLE
Excluding Tenants / Sites from doing certains Tests

### DISCUSSION THREAD

**Discussion by aascedu** [amyasnikov/validity#133] [Q&A]
Hi !

I was wondering if it could be possible to exclude a Tenant or a Site from doing certains tests.
For exemple, for a Cisco Router, 2 tests :
    - I will check that the NTP Server is set accordingly to a value.
    - I will check that `no ip http server` and `no ip http secure-server` are set

For almost every Tenants, I will want to execute both tests, but for some clients which could have their own NTP server, I don't want to execute the NTP Server test as I know it will not be complient and it is not supposed to be anyway.

I thought of doing multiple selectors but it seems messy and not that easy to keep updated and clean. So I was wondering if anyone had an idea !

Thank you for your input !
Arthur

**Comment by amyasnikov** [comment]
Hey Arthur, currently there is no option to directly exclude something, so you should focus on **including** only those devices you need for the tests. For example, if you don't have plenty of devices, **tags** can be a good option. Just mark all your devices with an appropriate tag and then use it in a selector.

**Reply by aascedu** [reply]
Do you think it could be possible to implement a way to exclude devices for test?
Since I would like to have that possibility, I could try working on it !
I thought of tags, but I do have a lot of devices, something in the thousands for different clients and locations meaning different config context, etc... and I thought it would not be viable !
I know that making personnalized selector could be an option but I think it will be too much work just to not see failed test results we don't have to take in consideration

**Reply by amyasnikov** [reply]
Yeah, I think it's possible, but would be no so easy to implement. Right now Selector is connected to other related models (like Site or Tenant) through Many-To-Many relationship. So, adding an auxiliary boolean field like "exclude" to this m2m table would allow to model negative components of the Selector in the DB. The more difficult part is to make the right form for this.

**Comment by aascedu** [comment]
OK, this sounds a bit advanced for me, its getting late, I will think about
it tomorrow !
Thank you my friend
